README.md
# Read Me
- PID: A15854601
- Netlify Link: https://boisterous-fenglisu-ed077c.netlify.app/
- Github Repo Link: https://github.com/kaicheng0824/cse134-hw5 

# Files


# Modifications of old sites
Detailed in [changelog.md](changelog.md)



changelog.md
# Changes Made in HW4

- Updated more experiences in the professional section
- Added Google Analytics javascript tag
- Added downloadable pdf resume and transcript
- Added Reading list tab in personal
- Added external links to projects and writing
- Added links to github and linkedin in contactme.html
- Added SVG icons for github, linked, instagram, and twitter
- Adjust footer

styledpost.js
export function definePost() {
    class Post extends HTMLElement { }
    customElements.define('post-articles', Post, { extends: 'div'})

    class PostTitle extends HTMLElement { }
    customElements.define('post-title', PostTitle, { extends: 'div'})

    class PostDate extends HTMLElement { }
    customElements.define('post-date', PostDate, { extends: 'div'})

    class PostSummary extends HTMLElement { }
    customElements.define('post-summary', PostSummary, { extends: 'div'})
}

export function generatePostId() {
    return crypto.randomUUID();
}

/**
 * @typedef {{
 * title: string
 * date: string
 * summary: string
 * }} Post
 */


export const examplePost = {
    "title": "A Day in UCSD",
    "date": "09/17/2021",
    "summary": "Today was a great day"
};

/* Storage Layer
 * ============= */

/**
 * @returns the JSON map of books from local storage.
 */
export function loadPosts() {
    return JSON.parse(localStorage.getItem('posts')) || {};
}

/**
 * @param {{string: Book}} books a JSON map of id->book to put into local storage.
 */
function storePosts(posts) {
    localStorage.setItem('posts', JSON.stringify(posts));
}

export function insertPost(post) {
    const posts = loadPosts();
    const postId = generatePostId();

    posts[postId] = post;
    storePosts(posts);
    console.log('Error in insert');
    
    return postId;
}

export function selectPost(postId) {
    const posts = loadPosts();

    return posts[postId];
}

export function selectAllPosts() {
    const posts = loadPosts();
    return posts;
}

export function updatePost(postId, post) {
    const posts = loadPosts();

    posts[postId] = post;

    console.log(posts[postId]);

    storePosts(posts);
}

export function deletePost(postId) {
    const posts = loadPosts();

    //If it's not there, just return false. If it is there, delete it.
    if (!(postId in posts)) return false;
    delete posts[postId];
    
    storePosts(posts)
    return true;
}

/**
 * @param {string} [postId] id of post
 * @param {Post} [post]  post
 * @return {DocumentFragment}
 */

export function renderPost(postId,post){
    console.log('--------')
    console.log(postId)
    console.log(post)
    const template = document.getElementById('post-template')
    const postElement = template.content.cloneNode(true);
    postElement.children[0].dataset.postId = postId;
    postElement.children[0].querySelector('button').id = postId;

    const titleH1= postElement.querySelector("post-title > h2");
    titleH1.textContent = post['title'];

    const dateH2= postElement.querySelector("post-date > h3");
    dateH2.textContent = post['date']

    const summaryP= postElement.querySelector("post-summary > p");
    summaryP.textContent = post['summary'];

    return postElement
};

/**
 * 
 * @param {string} postId 
 * @param {Post|undefined} post 
 * @param {HTMLElement} container 
 */
 export function displayPost(postId, post, container) {
    // console.log('Error in display');
    const postElement = renderPost(postId,post);

    const existing = container.querySelector(`[data-post-id="${postId}"]`)
    if(existing) {
        existing.remove();
    }

    if(post) {
        container.appendChild(postElement);
    }

}

/**
 * 
 * @param {*} container 
 */
 export function redisplayAllPosts(container) {
    const posts = selectAllPosts();

    for (const [id, post] of Object.entries(posts)) {
        // console.log('Error in redisplay');
        displayPost(id, post, container);
    }
}



//export { definePost, generatePostId, examplePost, renderPost,displayPost};


// let posts = [{title:"title",date:"date",summary:"summary"}];



// document.addEventListener('DOMContentLoaded', function(){
//     let addBtn = document.getElementById("add");
//     let dialog = document.getElementById("dialog");
//     let submitAdd = document.getElementById("submitAdd");
//     let list = document.getElementById("list");
//     let title = document.getElementById('title');
//     let date = document.getElementById('date');
//     let summary = document.getElementById('summary');

//     addBtn.addEventListener('click',function() {
//         dialog.open = true;
//     });
    
//     submitAdd.addEventListener('click', function(){
//         posts[posts.length] = {
//             title: title.value,
//             date: date.value,
//             summary: summary.value
//         }
//         loadInfo();
//     });
// });

// export function loadInfo(){
//     let title = document.getElementById('title');
//     let date = document.getElementById('date');
//     let summary = document.getElementById('summary');
    
//     for(let i = 0; i < posts.length; i++) {
//         let strTemp = `<li> ${title.value}, ${date.value},${summary.value}</li>`
//         list.innerHTML = list.innerHTML + strTemp;
//     }
// }

customdialogs.js
document.addEventListener('DOMContentLoaded', function() {
    const alertBtn = document.getElementById('alert');
    const confirmBtn = document.getElementById('confirm');
    const promptBtn = document.getElementById('prompt');
    const safe_prompt = document.getElementById('safer_prompt');
    let ok = document.getElementById('ok');
    let cancel = document.getElementById('cancel');
    let input = document.getElementById('input');
    let dialog = document.getElementById("dialog");
    let text = document.getElementById('text');
    let result = document.getElementById('result');
    // let theResult = 'None';
    // let type = 'None';
    //let result_text = `${type} result is: ${theResult}`

    alertBtn.addEventListener('click',function() {
        dialog.open = true;
        text.innerHTML = "Alert is pressed!";
        cancel.style.display = "none";
        input.style.display = "none";
        ok.addEventListener('click',function(){
            dialog.open=false;
            result.style.display = "none";
        }); 
    });

    confirmBtn.addEventListener('click',function() {
        let theResult = 'None';
        let type = 'None';

        type = 'Confirm';
        dialog.open = true;
        cancel.style.display = "block";
        input.style.display = "none";
        text.innerHTML = "Do you confirm this";
        cancel.addEventListener('click',function(){
            dialog.open=false;
            theResult = 'False';
            let result_text = `${type} result is: ${theResult}`;
            result.innerHTML = result_text;
            result.style.display = "block";
        });
        ok.addEventListener('click',function(){
            dialog.open=false;
            theResult = 'True';
            let result_text = `${type} result is: ${theResult}`;
            result.innerHTML = result_text;
            result.style.display = "block";
        }); 
    });

    promptBtn.addEventListener('click',function() {
        let theResult = 'None';
        let type = 'None';

        type = 'Prompt';
        dialog.open = true;
        cancel.style.display = "block";
        input.style.display = "block";
        text.innerHTML = "Please Enter Something";

        ok.addEventListener('click',function(){
            dialog.open=false;
            theResult = document.getElementById("input").value;
            theResult = DOMPurify.sanitize(theResult, {FORBID_TAGS: ['style','script']} );
            if(theResult==='') {theResult = "Nothing Entered";}
            let result_text = `${type} result is: ${theResult}`;
            result.innerHTML = result_text;
            result.style.display = "block";
        }); 

        cancel.addEventListener('click',function(){
            dialog.open=false;
            theResult = 'Nothing Entered';
            let result_text = `${type} result is: ${theResult}`;
            result.innerHTML = result_text;
            result.style.display = "block";
        });
    });
})




blog.js
export function definePost() {
    class Post extends HTMLElement { }
    customElements.define('post-articles', Post, { extends: 'div'})

    class PostTitle extends HTMLElement { }
    customElements.define('post-title', PostTitle, { extends: 'div'})

    class PostDate extends HTMLElement { }
    customElements.define('post-date', PostDate, { extends: 'div'})

    class PostSummary extends HTMLElement { }
    customElements.define('post-summary', PostSummary, { extends: 'div'})
}

export function generatePostId() {
    return crypto.randomUUID();
}

/**
 * @typedef {{
 * title: string
 * date: string
 * summary: string
 * }} Post
 */


export const examplePost = {
    "title": "A Day in UCSD",
    "date": "09/17/2021",
    "summary": "Today was a great day"
};

/* Storage Layer
 * ============= */

/**
 * @returns the JSON map of books from local storage.
 */
export function loadPosts() {
    return JSON.parse(localStorage.getItem('posts')) || {};
}

/**
 * @param {{string: Book}} books a JSON map of id->book to put into local storage.
 */
function storePosts(posts) {
    localStorage.setItem('posts', JSON.stringify(posts));
}

export function insertPost(post) {
    const posts = loadPosts();
    const postId = generatePostId();

    posts[postId] = post;
    storePosts(posts);
    console.log('Error in insert');
    
    return postId;
}

export function selectPost(postId) {
    const posts = loadPosts();

    return posts[postId];
}

export function selectAllPosts() {
    const posts = loadPosts();
    return posts;
}

export function updatePost(postId, post) {
    const posts = loadPosts();

    posts[postId] = post;

    storePosts(posts);
}

export function deletePost(postId) {
    const posts = loadPosts();

    // If it's not there, just return false. If it is there, delete it.
    // if (!(postId in posts)) return false;
    delete posts[postId];
    
    storePosts(posts)
    return true;
}

/**
 * @param {string} [postId] id of post
 * @param {Post} [post]  post
 * @return {DocumentFragment}
 */

export function renderPost(postId,post){
    console.log('--------')
    console.log(postId)
    console.log(post)
    const template = document.getElementById('post-template')
    const postElement = template.content.cloneNode(true);
    postElement.children[0].dataset.postId = postId;
    // console.log(postElement.children[0]);
    // console.log(postElement.children[0].querySelector('button'));
    postElement.children[0].querySelector('button').id = postId;
    // console.log(postElement.children[0]);

    const titleH1= postElement.querySelector("post-title > h1");
    titleH1.textContent = post['title'];
    // console.log(post.title)
    // console.log(post)

    const dateH2= postElement.querySelector("post-date > h2");
    dateH2.textContent = post['date']

    const summaryP= postElement.querySelector("post-summary > p");
    summaryP.textContent = post['summary']

    return postElement
};

/**
 * 
 * @param {string} postId 
 * @param {Post|undefined} post 
 * @param {HTMLElement} container 
 */
 export function displayPost(postId, post, container) {
    console.log('Error in display');
    const postElement = renderPost(postId,post);

    const existing = container.querySelector(`[data-post-id="${postId}"]`)
    if(existing) {
        existing.remove();
    }

    if(post) {
        container.appendChild(postElement);
    }

}

/**
 * 
 * @param {*} container 
 */
 export function redisplayAllPosts(container) {
    const posts = selectAllPosts();

    for (const [id, post] of Object.entries(posts)) {
        console.log('Error in redisplay');
        displayPost(id, post, container);
    }
}


//export { definePost, generatePostId, examplePost, renderPost,displayPost};


// let posts = [{title:"title",date:"date",summary:"summary"}];



// document.addEventListener('DOMContentLoaded', function(){
//     let addBtn = document.getElementById("add");
//     let dialog = document.getElementById("dialog");
//     let submitAdd = document.getElementById("submitAdd");
//     let list = document.getElementById("list");
//     let title = document.getElementById('title');
//     let date = document.getElementById('date');
//     let summary = document.getElementById('summary');

//     addBtn.addEventListener('click',function() {
//         dialog.open = true;
//     });
    
//     submitAdd.addEventListener('click', function(){
//         posts[posts.length] = {
//             title: title.value,
//             date: date.value,
//             summary: summary.value
//         }
//         loadInfo();
//     });
// });

// export function loadInfo(){
//     let title = document.getElementById('title');
//     let date = document.getElementById('date');
//     let summary = document.getElementById('summary');
    
//     for(let i = 0; i < posts.length; i++) {
//         let strTemp = `<li> ${title.value}, ${date.value},${summary.value}</li>`
//         list.innerHTML = list.innerHTML + strTemp;
//     }
// }